### ISSUE TITLE
miniature's mobs are classified as Monsters

### ISSUE THREAD

**Issue by pantsguyapp** [noobanidus/miniatures#16]
They are targeted by iron golems or other auto hostile attacking mobs and despawned in peaceful.
Playing in 1.16 versions.

**Comment by noobanidus** [comment]
This is strange. Are you using Minecolonies?

**Comment by pantsguyapp** [comment]
> This is strange. Are you using Minecolonies?

No? I tested them in a world with just this mod and vanilla Minecraft.

Pls, hotfix for 1.16.

**Comment by pantsguyapp** [comment]
Why would Minecolonies affect the type of the mob? It only add npcs that auto attack hostiles, like iron golems, and they attacked the mini me because it is somehow classified as monsters.

**Comment by noobanidus** [comment]
#6

**Comment by noobanidus** [comment]
Miniatures *are* implemented as a hostile entity because that's what was requested and how they were generally coded. Changing that now would require a significant overhaul that I currently have no plans for; I'd suggest looking for another mod that suits your needs better.

**Comment by pantsguyapp** [comment]
> Miniatures _are_ implemented as a hostile entity because that's what was requested and how they were generally coded. Changing that now would require a significant overhaul that I currently have no plans for; I'd suggest looking for another mod that suits your needs better.

Yeah such as the original Openblocks, I don't remember the mini me from that mod ever be a hostile mob, why do you feel the need to change it so that it can attack players if configured even if that has never been a thing in the original mod.

**Comment by pantsguyapp** [comment]
Reminder that Minecolonies has these mobs called "mercenary" that also auto attack hostile mobs similar to the guards, and they will completely ignore the excluding list for this mod, yes I tested, so this is still an issue when played with Minecolonies.

**Comment by noobanidus** [comment]
> > Miniatures _are_ implemented as a hostile entity because that's what was requested and how they were generally coded. Changing that now would require a significant overhaul that I currently have no plans for; I'd suggest looking for another mod that suits your needs better.
> 
> Yeah such as the original Openblocks, I don't remember the mini me from that mod ever be a hostile mob, why do you feel the need to change it so that it can attack players if configured even if that has never been a thing in the original mod.

This mod was directly requested by a number of streamers who used them in this fashion. 

If you don't like it, you don't have to use it.

**Comment by noobanidus** [comment]
@pantsguyapp Apologies for getting combative, my stress levels were super high last night. 

To explain further, it was specifically requested to make the miniatures hostile as that's the primary use that multiple streamers used them for; I realize this isn't part of the original OpenBlocks system.

It might be possible to add a non-hostile mob, but due to real-life stress (multiple family member hospitalizations), it's just not something I can put on my agenda right now.

**Comment by pantsguyapp** [comment]
> @pantsguyapp Apologies for getting combative, my stress levels were super high last night.
> 
> To explain further, it was specifically requested to make the miniatures hostile as that's the primary use that multiple streamers used them for; I realize this isn't part of the original OpenBlocks system.
> 
> It might be possible to add a non-hostile mob, but due to real-life stress (multiple family member hospitalizations), it's just not something I can put on my agenda right now.

I am so sorry dude, I wasn't aware of your real life situation at the time, just take it easy, and deal with this issue how you feel is the best.

**Comment by noobanidus** [comment]
> > @pantsguyapp Apologies for getting combative, my stress levels were super high last night.
> > To explain further, it was specifically requested to make the miniatures hostile as that's the primary use that multiple streamers used them for; I realize this isn't part of the original OpenBlocks system.
> > It might be possible to add a non-hostile mob, but due to real-life stress (multiple family member hospitalizations), it's just not something I can put on my agenda right now.
> 
> I am so sorry dude, I wasn't aware of your real life situation at the time, just take it easy, and deal with this issue how you feel is the best.

Thanks so much, but don't apologize! I was the one that got snappy.

Unfortunately there'll be a lot of duping involved to get a non-hostile entity. Basically, the request were for streamers whose viewers had contributed to spawn a "mini" which usually attacked them; hence it was easier to extend a Mob rather than have it as a passive entity and then try to add attacks to it. I'll keep thinking on it to see what's possible without just completely duplicating the entity.

**Comment by pantsguyapp** [comment]
> > > @pantsguyapp Apologies for getting combative, my stress levels were super high last night.
> > > To explain further, it was specifically requested to make the miniatures hostile as that's the primary use that multiple streamers used them for; I realize this isn't part of the original OpenBlocks system.
> > > It might be possible to add a non-hostile mob, but due to real-life stress (multiple family member hospitalizations), it's just not something I can put on my agenda right now.
> > 
> > 
> > I am so sorry dude, I wasn't aware of your real life situation at the time, just take it easy, and deal with this issue how you feel is the best.
> 
> Thanks so much, but don't apologize! I was the one that got snappy.
> 
> Unfortunately there'll be a lot of duping involved to get a non-hostile entity. Basically, the request were for streamers whose viewers had contributed to spawn a "mini" which usually attacked them; hence it was easier to extend a Mob rather than have it as a passive entity and then try to add attacks to it. I'll keep thinking on it to see what's possible without just completely duplicating the entity.

Here's my attempt, I have seen rats mod have it so that iron golems will attack wild rats in the vicinity, while if they are tamed they will be ignored, no need to ever changing the entity typing, plus, it is all in their config file. Another problem however would be, how will you get mobs from other mods to auto targetting them, such as Mcol guards. For this, you can enforce a mob list that basically declare if a specific mob is hostile toward the MEs, then have players to fill in the mobs themselves, another approach for Mcol is to use their hostile mob list section and include the MEs, which players can choose to enable it. For the pets, if a mob attacking the owner, the dogs should go after it, and kill the ME.

**Comment by pantsguyapp** [comment]
> > > @pantsguyapp Apologies for getting combative, my stress levels were super high last night.
> > > To explain further, it was specifically requested to make the miniatures hostile as that's the primary use that multiple streamers used them for; I realize this isn't part of the original OpenBlocks system.
> > > It might be possible to add a non-hostile mob, but due to real-life stress (multiple family member hospitalizations), it's just not something I can put on my agenda right now.
> > 
> > 
> > I am so sorry dude, I wasn't aware of your real life situation at the time, just take it easy, and deal with this issue how you feel is the best.
> 
> Thanks so much, but don't apologize! I was the one that got snappy.
> 
> Unfortunately there'll be a lot of duping involved to get a non-hostile entity. Basically, the request were for streamers whose viewers had contributed to spawn a "mini" which usually attacked them; hence it was easier to extend a Mob rather than have it as a passive entity and then try to add attacks to it. I'll keep thinking on it to see what's possible without just completely duplicating the entity.

Does seems like a complete rewrite instead of just adding another peaceful "ME_Passive", I will leave it onto you then.

**Comment by pantsguyapp** [comment]
> > > @pantsguyapp Apologies for getting combative, my stress levels were super high last night.
> > > To explain further, it was specifically requested to make the miniatures hostile as that's the primary use that multiple streamers used them for; I realize this isn't part of the original OpenBlocks system.
> > > It might be possible to add a non-hostile mob, but due to real-life stress (multiple family member hospitalizations), it's just not something I can put on my agenda right now.
> > 
> > 
> > I am so sorry dude, I wasn't aware of your real life situation at the time, just take it easy, and deal with this issue how you feel is the best.
> 
> Thanks so much, but don't apologize! I was the one that got snappy.
> 
> Unfortunately there'll be a lot of duping involved to get a non-hostile entity. Basically, the request were for streamers whose viewers had contributed to spawn a "mini" which usually attacked them; hence it was easier to extend a Mob rather than have it as a passive entity and then try to add attacks to it. I'll keep thinking on it to see what's possible without just completely duplicating the entity.

I am a coding dunce but I still don't get it though, whats wrong with have it passive then add in an attacking AI? All I am seeing is a few more coding lines.
